doc/examples/bundle_group_registration.py

```diff
@@ -0,0 +1,128 @@
+"""
+=============================
+Groupwise Bundle Registration
+=============================
+
+This example explains how to coregister a set of bundles to a common space that
+is not biased by any specific bundle. This is useful when we want to align all
+the bundles but do not have a target reference space defined by an atlas.
+
+How it works
+============
+
+The bundle groupwise registration framework in DIPY relies on streamline linear
+registration (SLR) and an iterative process.
+
+In each iteration, bundles are shuffled and matched in pairs. Then, each pair
+of bundles are simultaneously moved to a common space in between both.
+
+After all pairs have been aligned, a group distance metric is computed as the
+mean pairwise distance between all bundle pairs. With each iteration, bundles
+get closer to each other and the group distance decreases.
+
+When the reduction in the group distance reaches a tolerance level the process
+ends.
+
+To reduce computational time, by default both registration and distance
+computation are performed on streamline centroids (obtained with Quickbundles).
+
+Example
+=======
+
+We start by importing and creating the necessary functions:
+"""
+
+import os
+import zipfile
+from time import sleep
+from dipy.align.streamlinear import groupwise_slr
+from dipy.data import get_fnames
+from dipy.io.streamline import load_tractogram
+from dipy.viz import window, actor
+
+import logging
+logging.basicConfig(level=logging.INFO)
+
+colors = [[0.91, 0.26, 0.35], [0.99, 0.50, 0.38], [0.99, 0.88, 0.57],
+          [0.69, 0.85, 0.64], [0.51, 0.51, 0.63]]
+
+
+def show_bundles(bundles, colors, show=True, fname=None):
+
+    scene = window.Scene()
+    scene.SetBackground(1., 1, 1)
+    for (i, bundle) in enumerate(bundles):
+        lines_actor = actor.streamtube(bundle, colors[i], linewidth=0.3)
+        lines_actor.RotateX(-90)
+        lines_actor.RotateZ(90)
+        scene.add(lines_actor)
+    if show:
+        window.show(scene)
+    if fname is not None:
+        sleep(1)
+        window.record(scene, n_frames=1, out_path=fname, size=(900, 900))
```

Comment:
Can you move this function to a new module? You can create a file streamline file in a `viz` module

Reply:
I agree it is a good idea. There are however other examples ([slr](https://dipy.org/documentation/1.5.0/examples_built/bundle_registration/#example-bundle-registration) or [recobundles](https://dipy.org/documentation/1.5.0/examples_built/bundle_extraction/#example-bundle-extraction)) with similar logic. We may want to update those examples as well later.

Reply:
Yes, for sure we will update all these examples.